src/charm.py

```diff
@@ -0,0 +1,664 @@
+#!/usr/bin/env python3
+
+"""Microcloud charm."""
+
+import json
+import logging
+import os
+import shutil
+import subprocess
+import tarfile
+import tempfile
+import time
+from typing import Dict, List, Union
+
+from charms.operator_libs_linux.v2.snap import SnapCache, SnapError, SnapState
+from charms.operator_libs_linux.v2.snap import install_local as snap_install_local
+from ops.charm import (
+    CharmBase,
+    ConfigChangedEvent,
+    InstallEvent,
+    RelationCreatedEvent,
+    RelationJoinedEvent,
+    StartEvent,
+    StopEvent,
+    UpdateStatusEvent,
+)
+from ops.framework import StoredState
+from ops.main import main
+from ops.model import (
+    ActiveStatus,
+    BlockedStatus,
+    MaintenanceStatus,
+    ModelError,
+    WaitingStatus,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class MaasMicrocloudCharmCharm(CharmBase):
```

Comment:
```suggestion
class MaasMicroCloudCharm(CharmBase):
```

Looks like `Charm` is doubled here.